src/NBomber/Domain/Domain.fs

```diff
@@ -62,221 +50,228 @@ type StepListenerChannel() =
+    StepName: StepName
+    Listeners: StepListenerChannel
+}
 
-module internal Step =        
+type Step =
+    | Request  of RequestStep
+    | Listener of ListenerStep
+    | Pause    of TimeSpan
+    interface IStep
 
+module Step =
     let isRequest (step)  = match step with | Request _  -> true | _ -> false
     let isListener (step) = match step with | Listener _ -> true | _ -> false
-    let isPause (step)    = match step with | Pause _    -> true | _ -> false    
-    
-    let getName (step) = 
+    let isPause (step)    = match step with | Pause _    -> true | _ -> false
+
+    let getName (step) =
         match step with
         | Request s  -> s.StepName
         | Listener s -> s.StepName
         | Pause t    -> "pause"
-        
+
     let getRequest (step) =
         match step with
         | Request r -> r
         | _         -> failwith "step is not a Request"
 
-    let getListener (step) = 
-        match step with 
+    let getListener (step) =
+        match step with
         | Listener l -> l
         | _          -> failwith "step is not a Listener"
 
-    let execStep (step: Step, req: Request, timer: Stopwatch) = task {        
-        timer.Restart()        
+    let execStep (step: Step, req: Request, timer: Stopwatch) = task {
+        timer.Restart()
         match step with
         | Request r  -> let! resp = r.Execute(req)
                         timer.Stop()
                         let latency = Convert.ToInt64(timer.Elapsed.TotalMilliseconds)
-                        return (resp, latency)        
-        
+                        return (resp, latency)
+
         | Listener l -> let listener = l.Listeners.Get(req.CorrelationId)
                         let! resp = listener.GetResponse()
                         timer.Stop()
                         let latency = Convert.ToInt64(timer.Elapsed.TotalMilliseconds)
                         return (resp, latency)
-        
+
         | Pause time -> do! Task.Delay(time)
                         return (Response.Ok(req), int64(0))
     }
 
     let runSteps (steps: Step[], correlationId: CorrelationId,
                   latencies: List<List<Response*Latency>>,
                   exceptions: List<Option<exn>*ExceptionCount>, ct: CancellationToken) = task {
-        
-        do! Task.Delay(10)        
+
+        do! Task.Delay(10)
         let timer = Stopwatch()
 
         while not ct.IsCancellationRequested do
-            
+
             let mutable request = { CorrelationId = correlationId; Payload = null }
             let mutable skipStep = false
             let mutable stepIndex = 0
 
             for st in steps do
                 if not skipStep then
                     try
-                        let! (response,latency) = Step.execStep(st, request, timer)
-                            
-                        if not(Step.isPause st) then                        
+                        let! (response,latency) = execStep(st, request, timer)
```

Comment:
What the benefit of removing Step? maybe better to rename exec, then you will have Step.exec

Reply:
I had to remove this due to refactoring. After re-shuffling the functions `Step` was being resolved to `NBomber.Contracts.Assertion.Step` which isn't correct. I did not want to cascade changes from this file to other files at the moment (may be later).